polyfactory/factories/pydantic_factory.py

```diff
@@ -501,7 +501,7 @@ def _create_model(cls, _build_context: PydanticBuildContext, **kwargs: Any) -> T
         :returns: An instance of type T.
 
         """
-        if _build_context.get("factory_use_construct"):
+        if cls._get_build_context(_build_context).get("factory_use_construct"):
```

Comment:
Оh, sorry, I somehow have missed covering those methods. 

Your proposed fix will help, but fail for `factory_use_construct=True`, so probably it is better to pass `build_context` along with `build_parameters` to the all missing methods like `handle_collection_type_coverage`, `get_field_value_coverage`.

Reply:
I've made these changes

Reply:
Is the call to `cls._get_build_context` necessary anymore? The build context being passed to this function should never be `None` right since it's never directly called?

Reply:
I suppose in the future it may be called with build context as `None` still so we would still want the `_get_build_context` call. If not then I can change the parameter to be non-optional?

Reply:
I think it's fine as it is.